framework/src/meshmodifiers/SideSetsBetweenSubdomains.C

```diff
@@ -52,26 +52,32 @@ SideSetsBetweenSubdomains::modify()
 
   MeshBase::const_element_iterator   el  = mesh.active_elements_begin();
   const MeshBase::const_element_iterator end_el = mesh.active_elements_end();
-  for (; el != end_el ; ++el)
+
+  for (unsigned int i = 0; i < master_ids.size(); ++i)
```

Comment:
Here you are looping through elements `master_ids.size() * paired_ids.size()` times. If the number is large, doing this will be costly. I will suggest you only have the element loop, and make both `master_ids` and `paired_ids` in type of `std::set<SubdomainID>`, then you can check if an element side is a subdomain interface by `master_ids.count(curr_subdomain) > 0` and `paired_ids.count(neighbor->subdomain_id()) > 0`.

Another thing we proposed in the issue but is not done here is to check if there is any side between the two subdomains. If there is none, we should do nothing instead of adding an empty new side set.